lms/djangoapps/verify_student/api.py

```diff
@@ -33,3 +40,76 @@ def send_approval_email(attempt):
     else:
         email_context = {'user': attempt.user, 'expiration_datetime': expiration_datetime.strftime("%m/%d/%Y")}
         send_verification_approved_email(context=email_context)
+
+
+def create_verification_attempt(user, name, status, expiration_datetime=None):
+    """
+    Create a verification attempt.
+
+    This method is intended to be used by IDV implementation plugins to create VerificationAttempt instances.
+
+    Args:
+        user (User): the user (usually a learner) performing the verification attempt
+        name (string): the name being ID verified
+        status (string): the initial status of the verification attempt
+        expiration_datetime (datetime, optional): When the verification attempt expires. Defaults to None.
+
+    Returns:
+        id (int): The id of the created VerificationAttempt instance
+    """
+    verification_attempt = VerificationAttempt.objects.create(
+        user=user,
+        name=name,
+        status=status,
+        expiration_datetime=expiration_datetime,
+    )
+
+    return verification_attempt.id
+
+
+def update_verification_attempt(attempt_id, name=None, status=None, expiration_datetime=None):
+    """
+    Update a verification attempt.
+
+    This method is intended to be used by IDV implementation plugins to update VerificationAttempt instances.
+
+    Arguments:
+        * attempt_id (int): the verification attempt id of the attempt to update
+        * name (string, optional): the new name being ID verified
+        * status (string, optional): the new status of the verification attempt
+        * expiration_datetime (datetime, optional): The new expiration date and time
+
+    Returns:
+        * None
+    """
+    try:
+        attempt = VerificationAttempt.objects.get(id=attempt_id)
+    except VerificationAttempt.DoesNotExist:
+        log.error(
+            f'VerificationAttempt with id {attempt_id} was not found '
+            f'when updating the attempt to status={status}',
+        )
+        raise
+
+    if name is not None:
+        attempt.name = name
+
+    if status is not None:
+        attempt.status = status
+
+        status_list = [attr for attr in dir(VerificationAttemptStatus) if not attr.startswith('__')]
```

Comment:
Nit (optional): FWIW, Python 3.11 has [StrEnum](https://docs.python.org/3.11/library/enum.html#enum.StrEnum) now, which might be a more natural way to express this.

Reply:
I attempted to use StrEnum for this, however that involved needing to change the `VerificationAttemptStatus` class itself, which in turn would require me to refactor quite a bit in this application (especially within tests) so this unfortunately doesn't seem worth it.

Reply:
a pattern I've used for this in the past is to give my constant 'enum' class a list property with valid values. Eg https://github.com/edx/edx-exams/blob/1b76cf33d8b0fbca6e15742cad8ab49f02960f1d/edx_exams/apps/core/statuses.py#L54

That, or can't we just use `VerificationAttempt.status_choices`?

Reply:
Oh I see what happened, I misread the docs and had `VerificationAttemptStatus` inherit from `Flag` instead of StrEnum 🤦. I implemented what you suggested, and I didn't have to make any migrations, and everything seems to work just fine. Thanks so much!